### PR TITLE
Add checkOverflow to blackbox exporter config

### PR DESCRIPTION
**Pull request by Conorbro**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+type SafeConfig struct {
+	sync.RWMutex
+	C *Config
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
```
Comment (brian-brazil) [inline config.go]: This isn't needed here

File: main.go [hunk 2]
```diff
@@ -34,22 +33,16 @@ import (
...
 	HTTP    HTTPProbe     `yaml:"http"`
 	TCP     TCPProbe      `yaml:"tcp"`
 	ICMP    ICMPProbe     `yaml:"icmp"`
 	DNS     DNSProbe      `yaml:"dns"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
 }
 
 type HTTPProbe struct {
```
Comment (brian-brazil) [inline main.go]: All of these should move to config.go

File: main.go [hunk 3]
```diff
@@ -34,22 +33,16 @@ import (
...
 type Module struct {
 	Prober  string        `yaml:"prober"`
 	Timeout time.Duration `yaml:"timeout"`
 	HTTP    HTTPProbe     `yaml:"http"`
 	TCP     TCPProbe      `yaml:"tcp"`
 	ICMP    ICMPProbe     `yaml:"icmp"`
 	DNS     DNSProbe      `yaml:"dns"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
```
Comment (brian-brazil) [inline main.go]: This isn't being checked

**Comment by Conorbro** [comment]
@brian-brazil

File: main.go [hunk 4]
```diff
@@ -120,6 +53,10 @@ func (sc *SafeConfig) reloadConfig(confFile string) (err error) {
 		return err
 	}
 
+	if err := checkOverflow(c.XXX, "config"); err != nil {
```
Comment (brian-brazil) [inline main.go]: This should be in the Unmarshall of the Config class

File: main.go [hunk 5]
```diff
@@ -172,6 +109,17 @@ func probeHandler(w http.ResponseWriter, r *http.Request, conf *Config) {
 	h.ServeHTTP(w, r)
 }
 
+func checkOverflow(m map[string]interface{}, ctx string) error {
```
Comment (brian-brazil) [inline main.go]: I'd keep this in config.go

**Comment by Conorbro** [comment]
@brian-brazil

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**